phpBB/config/default/container/services_messenger.yml

```diff
@@ -0,0 +1,57 @@
+parameters:
+    core.messenger_queue_file: '%core.cache_dir%queue.%core.php_ext%'
+
+services:
+    messenger.method_collection:
+        class: phpbb\di\service_collection
+        arguments:
+            - '@service_container'
+        tags:
+            - { name: service_collection, tag: messenger.method, class_name_aware: true }
+
+    messenger.method.base:
+        class: phpbb\messenger\method\base
+        shared: false
+        arguments:
+            - '@assets.bag'
+            - '@config'
+            - '@dispatcher'
+            - '@language'
+            - '@log'
+            - '@request'
+            - '@user'
+            - '@messenger.queue'
+            - '@path_helper'
+            - '@ext.manager'
+            - '@template.twig.extensions.collection'
+            - '@template.twig.lexer'
+            - '%core.template.cache_path%'
+            - '%core.php_ext%'
+
+    messenger.method.email:
+        class: phpbb\messenger\method\phpbb_email
+        shared: false
+        parent: messenger.method.base
+        calls:
+            - [reset, []]
+            - [set_transport, []]
+        tags:
+            - { name: messenger.method }
+
+    messenger.method.jabber:
+        class: phpbb\messenger\method\phpbb_jabber
+        shared: false
+        parent: messenger.method.base
+        calls:
+            - [reset, []]
```

Comment:
This is something like I've answered above about set_transport. The point is initializing things so they'd be the same through the whole service (and other services). Otherwise you never know the result.